lib/utils.js

```diff
@@ -229,6 +230,147 @@ const format = body => {
   return data;
 };
 
+/*
+ * @typedef Reference
+ * @type object
+ *
+ * @property {type} Reference.id - Id of the content type
+ * @property {attr} Reference.attr - Attribute Index
+ * @property {input} Reference.input - Input of the reference
+ * @property {length} Reference.length - Number of instances
+ * @property {ct.id} Reference.ct.id - Id of the referenced content type
+ * @property {ct.index} Reference.ct.index - Position of the reference content type
+ */
+
+/*
+ * Inititalizes reference
+ *
+ * @param {object} types - content types object
+ *
+ *
+ * @returns {array[Reference]} - content type with reference data
+ */
+const references = (types) => {
+  const cts = _.cloneDeep(types);
+  const refs = [];
+  const position = {};
+
+  // Saves position of each content type
+  cts.forEach((type, index) => {
+    if (!position.hasOwnProperty(type.id)) {
+      position[type.id] = index;
+    }
+  });
+
+  // Iterates through each content type and attributes
+  cts.map(ct => {
+    ct.attributes.map((attr, index) => {
+      let inputs;
+      let length;
+
+      // Get the number of instances if repeatable
+      if (Array.isArray(attr.inputs)) {
+        inputs = attr.inputs[0];
+        length = attr.inputs.length;
+      }
+      else {
+        inputs = attr.inputs;
+      }
+
+      // Iterate through each inputs
+      Object.keys(inputs).forEach(input => {
+        // Checks if input is a reference
+        if (inputs[input].hasOwnProperty('reference') && inputs[input].reference === true) {
+          // Throw error if contentType is not defined
+          if (!inputs[input].settings.hasOwnProperty('contentType')) {
+            throw new Error('Reference must have a content type');
+          }
+
+          // Throw error if contentType is not valid
+          if (!position.hasOwnProperty(inputs[input].settings.contentType)) {
```

Comment:
It will throw an error if the content type is not available.
So if there is any invalid string or empty content-type, it will throw an error